Getting started (#1805)
yewstack/yew: docs/getting-started/project-setup.md

description: Set yourself up for success
---

## Overview

> Your local development environment will need several tools to compile, build, package and debug your Yew application.

Reviewer: I think this makes it sound less foreboding?

```suggestion
Your local development environment will need a couple of tools to compile, build, package and debug your Yew application.
```

> ## Install Rust, Rustup and Cargo

Reviewer: ```suggestion
## Installing Rust
```

> To install Rust compiler, Rustup package manager and the Cargo build tool, follow the [official instructions](https://www.rust-lang.org/tools/install).

Reviewer: ```suggestion
To install Rust follow the [official instructions](https://www.rust-lang.org/tools/install).
```

:::important
The minimum supported Rust version (MSRV) for Yew is `1.45.0`. Older versions can cause unexpected issues accompanied by incomprehensible error messages.
You can check your toolchain version using `rustup show` (under "active toolchain") or alternatively `rustc --version`. To update your toolchain, run `rustup update`.
:::

## Install WebAssembly target

> Rust compiles source code for a specific target.  For WebAssembly, this target is called "wasm32-unknown-unknown".  The following instruction will add this target to your development environment.

Reviewer: ```suggestion
Rust can compile source codes for different "targets" (e.g. different processors). The compilation target for browser-based WebAssembly is called "wasm32-unknown-unknown".  The following command will add this target to your development environment.
```

`rustup target add wasm32-unknown-unknown`

## Install Trunk

Trunk is the recommended tool for managing deployment and packaging, and will be used throughout the documentation and examples.
See [Wasm Build Tools](./../more/wasm-build-tools.md) for more information on packaging and alternatives.

> `cargo install trunk`

Reviewer: ```suggestion
```
# note that this might take a while to install, because it compiles everything from scratch
# Trunk also provides prebuilt binaries for a number of major package managers
# See https://trunkrs.dev/#install for further details
cargo install trunk
```
```

## Install wasm-bindgen-cli

> Trunk uses the wasm-bindgen-cli to perform deployment and packaging, install using the following command.

Reviewer: ```suggestion
Trunk uses a tool called wasm-bindgen-cli to perform deployment and packaging, which should be installed using the following command.
```

> `cargo install wasm-bindgen-cli`

Reviewer: ```suggestion
```
cargo install wasm-bindgen-cli
```
```

Author: I am confused here, not seeing what change is proposed.  Are you suggesting that I remove the command to install the wasm-bindgen-cli?
